`ci`: Update Earlgrey test lists & add new regression run

AlexJones0 opened the pull request:
* Update the test list to match the latest changes to upstream `earlgrey_1.0.0` (remove a test target that was removed, add newly passing end-to-end ROM and ROM_EXT tests).
* Add a new regression to run the above ROM_EXT tests in parallel.
* Introduce a new timeout option to the OpenTitan regression run to support longer-running tests.
* Update some outdated documentation surrounding the CI regression runner.

AlexJones0 committed:
[ot] tests/opentitan: earlgrey-tests: Remove RSA test

The `rsa_3072_verify` test was removed from Upstream OpenTitan Earlgrey
due to being obsolete, but was previously expected to be passing here.
Remove it from the list of passing tests.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] docs/opentitan: regression: Update regression documentation

When the script was extended to support specifying different execution
environments and to collect all the Earlgrey test statuses together in
one place, I forgot to update the accompanying documentation.

Make sure that is up to date with this commit.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] tests/opentitan: earlgrey-tests: Add newly passing E2E ROM tests

With appropriate upstream OpenTitan support, we now expect a large
amount of ROM end-to-end tests to pass in QEMU. Add all of these tests
to the test list and mark them as expected passes to catch regressions.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] .github/workflows: ci: Add Earlgrey ROM_EXT CI regression

With upstream support, there are now a variety of ROM_EXT end-to-end
tests that we might expect to pass or at least be flaky. To ensure that
regressions on these more complicated tests are being caught, add a
regression run for the non-SiVal ROM_EXT environment, which most of
these tests use.

Since this is a separate job in the regression workflow it will be run
in parallel by a separate runner, and so should not increase existing CI
times (unless the time to execute all ROM_EXT test overshadows the time
for executing other existing execution environments).

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] scripts/opentitan: run-bazel-tests.sh: Add test timeout option

When testing a specific execution environment, you can set a maximum
test timeout to use for that environment.

For environments where tests are expected to be quick this can be used
to decrease CI times (tests using unimplemented features may timeout, so
this provides a smaller upper bound on test execution). For environments
where tests may take a while when executed in parallel on a standard
GitHub runner (maybe longer than 60 seconds), this can be used to
increase the timeout for just executions in that environment.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] tests/opentitan: earlgrey-tests: Add new passing ROM_EXT tests

Add newly passing ROM_EXT tests to be checked in Earlgrey regressions
now that these are properly supported in upstream OpenTitan. Uses the
increased timeout to ensure that even when running in parallel on slower
runners, these tests should pass most of the time.

Mark the couple of tests that are most likely to be flaky due to strict
timing checks in the ROM as such, to hopefully avoid CI flakiness.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

AlexJones0 committed:
[ot] tests/opentitan: earlgrey-tests: Add common flaky tests

Add a couple more OpenTitan tests that are commonly seen as being flaky
in the logs (previously were not expected to pass).

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

rivos-eblot commented on scripts/opentitan/run-bazel-tests.sh:
```diff
@@ -41,6 +41,9 @@ if [ ! "$qemu_exec_env" ]; then
   qemu_exec_env="qemu"
   echo "Using default 'qemu' tag to test all exec envs"
 fi
+if [ ! "$test_timeout" ]; then
```
nit pick: I think in POSIX shell it is not legit to test for the empty string using this syntax; you would need `[ -z "$test_timeout" ]`; the shebang being `sh`, that is _dash_ on Debian at least, the script would need to be POSIX-compliant.

AlexJones0 committed:
[ot] tests/opentitan: earlgrey-tests: Mark sigverify test flaky

We expect this test to pass, but recent upstream changes increase the
time to emulate this test by a lot and it will often take 2-5+ minutes
to execute depending on host performance and load.

Mark it flaky instead of removing it because it could theoretically pass
if the runner is fast enough, but it is not expected to pass with the
standard 60 second timeout we give.

Signed-off-by: Alex Jones <alex.jones@lowrisc.org>

jwnrt commented on tests/opentitan/data/earlgrey-tests.txt:
```diff
@@ -33,8 +33,8 @@ pass: //sw/device/silicon_creator/lib/sigverify:rsa_verify_functest_sim_qemu_rom
 pass: //sw/device/silicon_creator/lib/sigverify:sigverify_dynamic_functest_hardcoded_sim_qemu_sival_rom_ext
-pass: //sw/device/silicon_creator/lib/sigverify:sigverify_dynamic_functest_wycheproof_sim_qemu_rom_with_fake_keys
-pass: //sw/device/silicon_creator/lib/sigverify:sigverify_dynamic_functest_wycheproof_sim_qemu_sival_rom_ext
+flaky: //sw/device/silicon_creator/lib/sigverify:sigverify_dynamic_functest_wycheproof_sim_qemu_rom_with_fake_keys  # We expect this to pass, but it takes takes >> 60 seconds to run
+flaky: //sw/device/silicon_creator/lib/sigverify:sigverify_dynamic_functest_wycheproof_sim_qemu_sival_rom_ext  # We expect this to pass, but it takes takes >> 60 seconds to run
```
Do you want to increase the timeout for these on the OT side?

AlexJones0 replied:
Yes probably, but OT CI can take a long time to run and I wanted to address CI issues in QEMU first.
